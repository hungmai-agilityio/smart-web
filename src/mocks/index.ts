// Interfaces
import { END_POINT } from '@/constants';
import { IPortfolio, IProduct, IProject } from '@/interface';

export const itemDropdown = [
  { label: 'Option 1', value: 'option1' },
  { label: 'Option 2', value: 'option2' },
  { label: 'Option 3', value: 'option3' }
];

export const mockImages: IPortfolio[] = [
  {
    id: '1',
    name: 'Project One',
    image: '/background-hero.webp',
    date: '2020-03-10T09:30:00.000Z',
    detail: 'This is a detailed description of project one.',
    view: 120,
    favorite: 45,
    comment: 12
  },
  {
    id: '2',
    name: 'Project Two',
    image: '/background-hero.webp',
    date: '2020-03-10T09:30:00.000Z',
    detail: 'This is a detailed description of project two',
    view: 85,
    favorite: 32,
    comment: 7
  },
  {
    id: '3',
    name: 'Project Three',
    image: '/background-hero.webp',
    date: '2020-03-10T09:30:00.000Z',
    detail: 'This is a detailed description of project three',
    view: 445,
    favorite: 632,
    comment: 71
  }
];

export const mockProduct: IProduct[] = [
  {
    id: '1',
    name: 'Project One',
    image: '/background-hero.webp',
    view: 120,
    users: ['User A', 'User B']
  },
  {
    id: '2',
    name: 'Project Two',
    image: '/background-hero.webp',
    view: 85,
    users: ['User C', 'User D']
  },
  {
    id: '3',
    name: 'Project Three',
    image: '/background-hero.webp',
    view: 445,
    users: ['User E', 'User F']
  }
];

export const mockProject: IProject[] = [
  {
    id: '1',
    name: 'Project One',
    image: '/background-hero.webp',
    category: 'Category-A',
    comment: 1232,
    favorite: 444,
    views: 81532
  },
  {
    id: '2',
    name: 'Project Two',
    image: '/background-hero.webp',
    category: 'Category-A',
    comment: 1232,
    favorite: 444,
    views: 81532
  },
  {
    id: '3',
    name: 'Project Three',
    image: '/background-hero.webp',
    category: 'Category-A',
    comment: 1232,
    favorite: 444,
    views: 81532
  }
];

export const mockCategories = ['CategoryA', 'CategoryB', 'CategoryC', 'CategoryD']

