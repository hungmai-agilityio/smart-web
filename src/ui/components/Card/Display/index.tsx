import { memo } from 'react';

// Components
import { CardImage, StatsRow } from '@/ui/components';

interface CardDisplayProps {
  name: string;
  image: string;
  views: number;
  favorites: number;
  comments: number;
}

export const CardDisplay = memo(
  ({ name, image, views, favorites, comments }: CardDisplayProps) => {
    return (
      <div className="border rounded-lg shadow-md">
        <div className="w-full aspect-image relative">
          <CardImage src={image} alt={name} autoSize isBlur>
            <StatsRow
              views={views}
              favorite={favorites}
              comment={comments}
              viewIcon="/view-white.svg"
              favoriteIcon="/likes-white.svg"
              commentIcon="/cmt-white.svg"
              isColumn
            />
          </CardImage>
        </div>

        <div className="md:flex hidden items-center w-full justify-between py-3 text-x-xs">
          <StatsRow
            views={views}
            favorite={favorites}
            comment={comments}
            viewIcon="/view-dark.svg"
            favoriteIcon="/likes.svg"
            commentIcon="/comments.svg"
          />
        </div>
      </div>
    );
  }
);
