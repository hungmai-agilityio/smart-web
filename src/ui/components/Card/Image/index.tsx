'use client';
import { ReactNode, useState } from 'react';
import { clsx } from 'clsx';
import Image, { ImageProps } from 'next/image';

interface CardImageProps extends ImageProps {
  children?: ReactNode;
  styles?: string;
  isCircle?: boolean;
  fixedSize?: { width: number; height: number };
  autoSize?: boolean;
  isBlur?: boolean;
}

export const CardImage = ({
  src,
  alt,
  children,
  styles,
  isCircle,
  fixedSize,
  autoSize,
  isBlur,
  ...res
}: CardImageProps) => {
  const [hovered, setHovered] = useState(false);

  const toggleHover = () => setHovered(!hovered);

  const containerStyles = clsx('relative', styles, {
    'rounded-full overflow-hidden border-4 border-gray-300 hover:border-second':
      isCircle,
    'w-full h-full': autoSize,
    [`w-[${fixedSize?.width}px] h-[${fixedSize?.height}px]`]:
      fixedSize && !autoSize
  });

  return (
    <div
      className={containerStyles}
      onMouseEnter={toggleHover}
      onMouseLeave={toggleHover}
    >
      <Image
        src={src}
        alt={alt}
        fill={autoSize}
        width={fixedSize?.width}
        height={fixedSize?.height}
        style={{ objectFit: 'cover' }}
        className={clsx('transition-opacity duration-300', {
          'opacity-50': hovered && children
        })}
        placeholder={isBlur ? 'blur' : 'empty'}
        blurDataURL={
          isBlur
            ? 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPkNA3fBAACLQFJQfEBVAAAAABJRU5ErkJggg=='
            : undefined
        }
        {...res}
      />
      {hovered && children && (
        <div className="absolute inset-0 flex items-center bg-black bg-opacity-50 text-white p-4">
          {children}
        </div>
      )}
    </div>
  );
};
