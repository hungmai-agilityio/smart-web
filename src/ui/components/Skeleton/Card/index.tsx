export const CardSkeleton = () => {
  return (
    <div className="grid md:grid-cols-4 grid-cols-1 w-5/6 gap-6 mx-auto">
      {Array.from({ length: 4 }).map((_, index) => (
        <div key={index} className="flex flex-col animate-pulse">
          <div className="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto bg-gray-300 rounded-full" />

          <div className="mt-4 md:h-32 text-center">
            <div className="h-6 bg-gray-300 rounded w-3/4 mx-auto mb-2"></div>
            <div className="h-4 bg-gray-300 rounded w-1/4 mx-auto mb-2"></div>
            <div className="h-4 bg-gray-300 rounded w-2/4 mx-auto"></div>
          </div>

          <div className="mt-5 mx-auto bg-gray-300 rounded w-24 h-8"></div>
        </div>
      ))}
    </div>
  );
};
