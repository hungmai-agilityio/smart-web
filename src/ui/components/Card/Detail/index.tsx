import { memo } from 'react';
import Image from 'next/image';

// Constants
import { montserrat, TYPE } from '@/constants';

// Components
import { Button, CardImage } from '@/ui/components';
import { clsx } from 'clsx';

interface CardDetailProps {
  id: string;
  name: string;
  image: string;
  users?: string[];
  view: number;
  nameButton?: string;
  onClick?: (id: string) => void;
}

export const CardDetail = memo(
  ({ id, name, image, users, view, nameButton, onClick }: CardDetailProps) => {
    const handleClick = () => {
      if (onClick) {
        onClick(id);
      }
    };

    return (
      <div className="flex flex-col">
        <div className="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto">
          <CardImage src={image} alt={name} autoSize isCircle isBlur>
            <div className="w-full">
              <Image
                src="/view.svg"
                alt="view icon"
                className="md:block hidden mx-auto"
                width={40}
                height={22}
              />
              <div className="flex justify-center items-center gap-3 md:mt-10 mt-0">
                <Image src="/view.svg" alt="view icon" width={22} height={11} />
                <span className="text-sm">{view}</span>
              </div>
            </div>
          </CardImage>
        </div>
        <div
          className={clsx('mt-4 text-center md:h-32 overflow-hidden', {
            'md:h-7': !users
          })}
        >
          <p
            className={`${montserrat.className} font-bold lg:text-lg text-base`}
          >
            {name}
          </p>
          {users && (
            <>
              <span className="my-3">by</span>
              <p className="lg:text-base text-sm mt-3">{users?.join(', ')}</p>
            </>
          )}
        </div>
        {nameButton && (
          <Button
            variant={TYPE.SECOND}
            className="mt-5 mx-auto"
            onClick={handleClick}
          >
            {nameButton}
          </Button>
        )}
      </div>
    );
  }
);
