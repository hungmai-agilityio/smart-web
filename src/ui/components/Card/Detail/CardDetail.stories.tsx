import type { Meta, StoryObj } from '@storybook/react';

// Components
import { CardDetail } from '@/ui/components/Card/Detail';

// Mocks
import { mockProduct } from '@/mocks';

const meta: Meta<typeof CardDetail> = {
  component: CardDetail,
  tags: ['autodocs'],
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    id: {
      description: 'Data id'
    },
    image: {
      description: 'Image of item for src'
    },
    name: {
      description: 'Name for the image when the image is corrupted'
    },
    users: {
      description: 'List user'
    },
    nameButton: {
      description: 'Name of the button'
    },
    onClick: {
      description: 'Handles button clicks to get item information'
    },
    view: {
      description: 'Total number of people who viewed the item'
    }
  }
};

export default meta;

type Story = StoryObj<typeof CardDetail>;

const handleClick = (id: string) => {
  alert(`Clicked on ${id}`);
};

export const Default: Story = {
  args: {
    id: mockProduct[0].id,
    name: mockProduct[0].name,
    image: mockProduct[0].image,
    users: mockProduct[0].users,
    view: mockProduct[0].view
  }
};

export const HasButton: Story = {
  args: {
    id: mockProduct[0].id,
    name: mockProduct[0].name,
    image: mockProduct[0].image,
    users: mockProduct[0].users,
    view: mockProduct[0].view,
    nameButton: 'Review',
    onClick: handleClick
  }
};
