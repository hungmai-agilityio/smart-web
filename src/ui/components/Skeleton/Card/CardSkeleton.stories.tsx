import { Meta, StoryObj } from '@storybook/react';
import { CardSkeleton } from '@/ui/components/Skeleton/Card';

const meta: Meta<typeof CardSkeleton> = {
  component: CardSkeleton,

  parameters: {
    nextjs: {
      appDirectory: true
    }
  }
};

export default meta;

type Story = StoryObj<typeof CardSkeleton>;

export const Default: Story = {
  args: {}
};
