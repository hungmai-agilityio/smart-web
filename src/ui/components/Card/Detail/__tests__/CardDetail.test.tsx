import { render } from '@testing-library/react';

// Components
import { CardDetail } from '@/ui/components/Card/Detail';

// Mocks
import { mockProduct } from '@/mocks';

describe('CardDetail', () => {
  const defaultProps = {
    id: mockProduct[0].id,
    name: mockProduct[0].name,
    image: mockProduct[0].image,
    users: mockProduct[0].users,
    view: mockProduct[0].view
  };

  it('Should match snapshot', () => {
    const { container } = render(<CardDetail {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
