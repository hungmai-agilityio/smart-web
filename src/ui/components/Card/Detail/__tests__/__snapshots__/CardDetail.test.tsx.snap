// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetail Should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <div
      class="xl:w-pro-lg xl:h-pro-lg md:w-pro-md md:h-pro-md w-pro-sm h-pro-sm mx-auto"
    >
      <div
        class="relative rounded-full overflow-hidden border-4 border-gray-300 hover:border-second w-full h-full"
      >
        <img
          alt="Project One"
          class="transition-opacity duration-300"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fbackground-hero.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fbackground-hero.webp&w=640&q=75 640w, /_next/image?url=%2Fbackground-hero.webp&w=750&q=75 750w, /_next/image?url=%2Fbackground-hero.webp&w=828&q=75 828w, /_next/image?url=%2Fbackground-hero.webp&w=1080&q=75 1080w, /_next/image?url=%2Fbackground-hero.webp&w=1200&q=75 1200w, /_next/image?url=%2Fbackground-hero.webp&w=1920&q=75 1920w, /_next/image?url=%2Fbackground-hero.webp&w=2048&q=75 2048w, /_next/image?url=%2Fbackground-hero.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        />
      </div>
    </div>
    <div
      class="mt-4 text-center md:h-32 overflow-hidden"
    >
      <p
        class="className font-bold lg:text-lg text-base"
      >
        Project One
      </p>
      <span
        class="my-3"
      >
        by
      </span>
      <p
        class="lg:text-base text-sm mt-3"
      >
        User A, User B
      </p>
    </div>
  </div>
</div>
`;
